app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -605,7 +605,22 @@ export default class Heatmap {
       this.clusterRows();
     }
 
-    if (this.columnMetadataSortField) return;
+    if (this.columnMetadataSortField) {
+      this.columnClustering = null;
+      orderBy(
+        this.columnLabels,
+        label => {
+          return (
+            (label.metadata && label.metadata[this.columnMetadataSortField]) ||
+            "ZZZ"
+          );
+        },
+        this.columnMetadataSortAsc ? "asc" : "desc"
+      ).forEach((label, idx) => {
+        label.pos = idx;
+      });
+      return;
```

Comment:
Should remove the return and turn the next if into an `else if`.

Reply:
Sounds good, done!